В начале как всегда настроим окружение: импортируем все необходимые библиотеки и немного настроим дефолтное отображение картинок.

In [1]:
# Python 2 and 3 compatibility
# pip install future
from __future__ import (absolute_import, division,
                        print_function, unicode_literals)
# отключим предупреждения Anaconda
import warnings
warnings.simplefilter('ignore')

# будем отображать графики прямо в jupyter'e
%pylab inline

#увеличим дефолтный размер графиков
from pylab import rcParams
rcParams['figure.figsize'] = 8, 5
import pandas as pd
import seaborn as sns

Populating the interactive namespace from numpy and matplotlib


После этого загрузим в DataFrame данные, с которыми будем работать. Для примеров я выбрала данные о продажах и оценках видео-игр из Kaggle Datasets.

In [ ]:
df = pd.read_csv('Games.csv')
# df.info()

In [20]:
df.head(1)

Name Platform  Year_of_Release   Genre Publisher  NA_Sales  EU_Sales  \
0  Wii Sports      Wii           2006.0  Sports  Nintendo     41.36     28.96   

   JP_Sales  Other_Sales  Global_Sales  Critic_Score  Critic_Count User_Score  \
0      3.77         8.45         82.53          76.0          51.0          8   

   User_Count Developer Rating  
0       322.0  Nintendo      E

Некоторые признаки, которые pandas считал как object, явно приведем к типам float или int.

In [49]:
df['User_Score'] = df.User_Score.astype('float64')
df['Year_of_Release'] = df.Year_of_Release.astype('int64')
df['User_Count'] = df.User_Count.astype('int64')
df['Critic_Count'] = df.Critic_Count.astype('int64')
# df.info()
# df[df['User_Score'] > 8]['Name'].count()

2125

Данные есть не для всех игр, поэтому давайте оставим только те записи, в которых нет пропусков, с помощью метода dropna.

In [27]:
df = df.dropna()
print(df.shape)

(6825, 16)


Всего в таблице 6825 объектов и 16 признаков для них. Посмотрим на несколько первых записей c помощью метода head, чтобы убедиться, что все распарсилось правильно. Для удобства я оставила только те признаки, которые мы будем в дальнейшем использовать.

In [50]:
useful_cols = ['Name', 'Platform', 'Year_of_Release', 'Genre', 
               'Global_Sales', 'Critic_Score', 'Critic_Count',
               'User_Score', 'User_Count', 'Rating'
              ]
df[useful_cols].head(100)

Name Platform  Year_of_Release  \
0                                      Wii Sports      Wii             2006   
2                                  Mario Kart Wii      Wii             2008   
3                               Wii Sports Resort      Wii             2009   
6                           New Super Mario Bros.       DS             2006   
7                                        Wii Play      Wii             2006   
8                       New Super Mario Bros. Wii      Wii             2009   
11                                  Mario Kart DS       DS             2005   
13                                        Wii Fit      Wii             2007   
14                             Kinect Adventures!     X360             2010   
15                                   Wii Fit Plus      Wii             2009   
16                             Grand Theft Auto V      PS3             2013   
17                  Grand Theft Auto: San Andreas      PS2             2004   
19   Brain Age: Train Your Brain in Minutes a Day       DS             2005   
23                             Grand Theft Auto V     X360             2013   
24                    Grand Theft Auto: Vice City      PS2             2002   
26    Brain Age 2: More Training in Minutes a Day       DS             2005   
28                         Gran Turismo 3: A-Spec      PS2             2001   
29                 Call of Duty: Modern Warfare 3     X360             2011   
32                        Call of Duty: Black Ops     X360             2010   
34                     Call of Duty: Black Ops II      PS3             2012   
35                     Call of Duty: Black Ops II     X360             2012   
36                 Call of Duty: Modern Warfare 2     X360             2009   
37                 Call of Duty: Modern Warfare 3      PS3             2011   
38                           Grand Theft Auto III      PS2             2001   
39                        Super Smash Bros. Brawl      Wii             2008   
40                                   Mario Kart 7      3DS             2011   
41                        Call of Duty: Black Ops      PS3             2010   
42                             Grand Theft Auto V      PS4             2014   
43                    Animal Crossing: Wild World       DS             2005   
44                                         Halo 3     X360             2007   
..                                            ...      ...              ...   
106                                      Tekken 3       PS             1998   
107        The Legend of Zelda: Twilight Princess      Wii             2006   
109                     Crash Bandicoot 3: Warped       PS             1998   
110                                  Mario Kart 8     WiiU             2014   
111                       Super Smash Bros. Melee       GC             2001   
113                  Need for Speed Underground 2      PS2             2004   
114                     Medal of Honor: Frontline      PS2             2002   
116                                  Just Dance 4      Wii             2012   
117                                Gears of War 2     X360             2008   
118                Uncharted 3: Drake's Deception      PS3             2011   
120                Call of Duty 4: Modern Warfare      PS3             2007   
121                    Uncharted 2: Among Thieves      PS3             2009   
123                             Big Brain Academy       DS             2005   
125                           Red Dead Redemption      PS3             2010   
126                                       FIFA 14      PS3             2013   
127                          Assassin's Creed III      PS3             2012   
128                   Donkey Kong Country Returns      Wii             2010   
129                          Halo: Combat Evolved       XB             2001   
130                   The Elder Scrolls V: Skyrim      PS3             2011   
131             